src/picongpu/include/plugins/adios/ADIOSWriter.hpp

```diff
@@ -662,6 +665,22 @@ class ADIOSWriter : public ILightweightPlugin
             }
         }
 
+
+        if( lastSpeciesSyncStep != currentStep )
+        {
+            /*copy species only one time per timestep to the host*/
+
+            DataConnector &dc = Environment<>::get().DataConnector();
+            dc.getData<MallocMCBuffer> (MallocMCBuffer::getName());
+
+            /* we copy all species to the host because we can not check if
```

Comment:
```c++
/* here we are copying all species to the host side since we
 * can not say at this point if this time step will need all of them
 * for sure (checkpoint) or just some user-defined species (dump) */
```